### PR TITLE
7747 docker ps

### PR DESCRIPTION
**Pull request by p4tin**
This changes makes the default ps output only 4 columns and reflects the wishes of 95% of the comments in the ticket #7477.  I have also updated the tests in integration-cli to refect these changes.

Signed-off-by: Paul Fortin paul@cp-soft.com

### CODE REVIEW

File: api/client/ps.go [hunk 1]
```diff
@@ -173,3 +224,12 @@ func (cli *DockerCli) CmdPs(args ...string) error {
 
 	return nil
 }
+
+func stringInSlice(a string, list []string) bool {
```
Comment (runcom) [inline api/client/ps.go]: There's just something similar in `pkg/stringutils` at https://github.com/docker/docker/blob/master/pkg/stringutils/stringutils.go#L44

File: runconfig/parse.go [hunk 2]
```diff
@@ -101,12 +99,7 @@ func Parse(cmd *flag.FlagSet, args []string) (*Config, *HostConfig, *flag.FlagSe
 		return nil, nil, cmd, err
 	}
 
-	// Validate input params
-	if *flWorkingDir != "" && !path.IsAbs(*flWorkingDir) {
```
Comment (runcom) [inline runconfig/parse.go]: I may not fully understand but why you have removed this?

**Comment by duglin** [comment]
please squash your commits

**Comment by duglin** [comment]
I think this is a dup of: #10255 - or at least close enough that I'd prefer if you worked with that PR to see if it can be morphed into what you need.

**Comment by cpuguy83** [comment]
Looks like you have a commit here that shouldn't be there.

**Comment by cpuguy83** [comment]
Closing this one since there is already #10255 as @duglin mentioned.
Also we can't change the default layout of this command.
This is in affect an API that people rely on that we are essentially stuck with until we are ready to make some breaking changes.

**Comment by cpuguy83** [comment]
Also thanks for taking a stab at this @p4tin

**Comment by p4tin** [comment]
This is cool - was my first stab at a change for docker (or any other open
source project for that matter) - I am learning a lot and want to keep
going - I picked this one because no one else had #dibs'ed it and that was
the process laid out in the docs - since I am new I have a lot of
questions, is there a chat / forum / etc,... where I can ask newbie
questions?

As far as the change for this PR, it did not change the layout if you
specified the columns in the command: --columns="id,..." would produce the
exact same layout as you had for backwards compatibility sake.  Was just
trying to get the change as most had requested it to look.

I just looked at #10255 and that seems like a good way to go as well with
just letters but more cryptic for the lay user.

Thanks for all the constructive comments, I will pick a new change to work
on (any suggestions?)

**Comment by cpuguy83** [comment]
@p4tin Check out https://docs.docker.com/project/find-an-issue/ for a good way to find an issue to working that's (hopefully) suitable for you.

**Pull request closed without merging**